docs/explanation.rst

```diff
@@ -0,0 +1,121 @@
+Explanantion
+=============
+
+..
+  This section requires more improvements.
+
+
+Upload Sets
+-----------
+``UploadSet`` is The main object provided by this extension. Each `UploadSet` is
+responsible for dealing with set of files. You declare `UploadSet` in the code::
+
+    photos = UploadSet('photos', IMAGES)
+
+And then you can use the `UploadSet.save` method to save uploaded files and
+`UploadSet.path` and `UploadSet.url` to access them.
+
+during saving & retrieving the uploaded file, `UploadSet` perform some security
+checks for you.
+
+
+Security Checks:
+----------------
+
+Security itself is a huge subject and you should consider designing it
+seriously. This extension is not intended by first intention to protect your
+app. However ``Flask-Reuploaded`` extension tries to help you in preveting the
+common simple attacks or user errors. for Example::
+
+    - User can't upload file with filename that starts with `.` 
+      This means that user can't upload `.`, '..', '../../../home/{username}/.bash' or '../../my_app.wsgi'.
+      i.e:. filename will cahnged or internal server error will be raised.
+    - User can't overwrite other users files i.e:. `similar filenames are suffixed by number`
+    - User can't upload file with extension suffix that is not allowed by the ``UploadSet``.
+
+You are adviced to read the source code to explore the mechanism of dealing with
+each error/attack and you may decide to fix something yourself.
+
+Another security aspect that you should consider is the maximum allowed file
+size. This is ``NOT`` managed by `Flask-Reuploaded` extension nor the `Flask`
+itself.
+
+By default `Flask` will happily accept one hundred GB file upload till crashing your hard
+drive by `no empty space error`. You can set maximum length by setting
+``MAX_CONTENT_LENGTH`` configuration. after this `flask` will reject files that
+are more than this limit.
+
+This is alone may not be enough. As uploading one thousand files of 10 MB length
+will crash your server also. This aspects is out of the scope of this extension
+but worth mentioned here.
+
+
+
+
+Default Dest Callable Parameter:
+--------------------------------
+
+`UploadSet` constructor can accept ``default_dest`` parameter, it is callable
+that takes the application as its argument & returns the destination path for
+this set.
+
+For example::
+
+    .. code-block:: python
+
+        media = UploadSet('media', default_dest=lambda app: os.path.join(app.instance_path, "uploads"))
```

Comment:
Could you make this a two liner? And pull out the lambda expression? Let's not make the code examples harder to understand for newcomers than necessary.